sdks/python/apache_beam/yaml/yaml_transform.py

```diff
@@ -955,10 +955,37 @@ def preprocess_languages(spec):
     else:
       return spec
 
+  def validate_transform_references(spec):
+    if 'transforms' not in spec:
+      return spec
+
+    for transform in spec['transforms']:
+      name = transform.get('name')
+      inputs = transform.get('input')
+      if name is None or inputs is None:
```

Comment:
What about references to self by type? Should we lent name = `transform.get('name', tranform.get('type'))`?

Reply:
I was thinking of validating input against both name and type to address cases like below.  

```
type: composite
    transforms:
      - type: Create
        name: CreateData
        config:
            elements: [0, 1, 3, 4]
        input: Create
      - type: PyMap
        name: PyMap
        config:
            fn: "lambda row: row.element * row.element"
        input: Create
    output: PyMap
```

A clarification : Can a transform's input reference another Named transform by type ?

Reply:
@robertwb As time permits, can you plz help clarify. I will send the updated PR accordingly.

Reply:
Yes. Essentially the name defaults to type if none is explicitly given.